packages/zoe/src/zoeService/feeMint.js

```diff
@@ -0,0 +1,42 @@
+// @ts-check
+
+import { makeIssuerKit } from '@agoric/ertp';
+
+import { makeHandle } from '../makeHandle.js';
+
+const { details: X } = assert;
+/**
+ * @param {FeeIssuerConfig} feeIssuerConfig
+ * @returns {{
+ *    feeMintAccess: FeeMintAccess,
+ *    getFeeIssuerKit: GetFeeIssuerKit,
+ *    feeIssuer: Issuer }}
+ */
+const createFeeMint = feeIssuerConfig => {
+  /** @type {IssuerKit} */
+  const feeIssuerKit = makeIssuerKit(
+    feeIssuerConfig.name,
+    feeIssuerConfig.assetKind,
+    feeIssuerConfig.displayInfo,
+  );
```

Comment:
The call to makeIssuerKit in `zoeStorageManager` includes an optShutdownWithFailure` argument, that shuts down the appropriate zcf vat if a consistency-loss error occurs. IIUC, this feeIssuer is zoe-wide rather than per zcf, so we can't make the error that local. But we should still shut down something, like perhaps the zoe service vat itself, rather than risking that we continue execution after losing consistency.

```suggestion
  const feeIssuerKit = makeIssuerKit(
    feeIssuerConfig.name,
    feeIssuerConfig.assetKind,
    feeIssuerConfig.displayInfo,
    shutdownSomethingAppropriate,
  );
```

Reply:
`makeZoeKit` now takes a `shutdownZoeVat` function which is passed to `makeIssuerKit` for the fee issuerKit and the invitation issuerKit.